### ISSUE TITLE
Hans The Almighty crashing game in Nether [Ticking Entity]

### ISSUE THREAD

**Issue by Zecheariah** [AnonymousHacker1279/ImmersiveWeapons#63]
~~When exploring in the Nether~~ My game and/or the server I'm running crashes because of Hans The Almighty. It's happened multiple times and I've found no way to fix it(Other than running a repeating command killing every Hans The Almighty). It happens on the server and single player.

To reproduce this bug, you just have to float around ~~in the nether~~ anywhere and then out of nowhere the game crashes. Probably because a Hans The Almighty Spawns in and crashes the game

Expected Behavior is a ticking entity crash.

Setup:
 - Forge 36.2.20
 - Immersive Weapons 1.4.2
 - mods folder from mod pack
[modlist.txt](https://github.com/AnonymousHacker1279/ImmersiveWeapons/files/7810395/modlist.txt)

 I've tried deleting the entirety of the nether dimension and looking in the immersive weapons config file to disable spawning of Hans but there are only structure options in the config file. I thought it was only in the nether since I've only crashed in the Nether but it's actually in the overworld as well when I look closely at the crash reports.
 
 If anything else is needed I can provide it. 
 
 
 Hans The Almighty in another location crash:
 [crash-2022-01-01_14.10.29-server.txt](https://github.com/AnonymousHacker1279/ImmersiveWeapons/files/7810416/crash-2022-01-01_14.10.29-server.txt)
 
Hans The Almighty in the Nether:
[crash-2022-01-02_12.07.00-server.txt](https://github.com/AnonymousHacker1279/ImmersiveWeapons/files/7810475/crash-2022-01-02_12.07.00-server.txt)

 Most recent crash:
[crash-2022-01-04_13.45.06-server.txt](https://github.com/AnonymousHacker1279/ImmersiveWeapons/files/7810385/crash-2022-01-04_13.45.06-server.txt)

**Comment by AnonymousHacker1279** [comment]
All 1.16 versions of Immersive Weapons are at End of Life and are not receiving updates. Chances are it has been fixed in v1.5.x+ (1.17 versions). However, I'll take a look into it, if it's an easy enough fix then I'll release an update. Can you upload your debug log? It'll be under the `logs` folder of your game directory.

As for the config file, versions before 1.9.3 do not support configuring the natural spawning of entities.

**Comment by Zecheariah** [comment]
It's understandable if you can't fix it, I can just run a command to kill Hans when he spawns or just remove the mod. Thanks for even looking into it :)

Here's a crash that JUST happened for this issue:
[crash-2022-01-04_16.04.06-server.txt](https://github.com/AnonymousHacker1279/ImmersiveWeapons/files/7810657/crash-2022-01-04_16.04.06-server.txt)

Here is the log for said crash.
[2022-01-04-10.log.gz](https://github.com/AnonymousHacker1279/ImmersiveWeapons/files/7810654/2022-01-04-10.log.gz)
.

**Comment by AnonymousHacker1279** [comment]
I would seriously enjoy being able to support multiple versions, but unfortunately, I do not have enough time or resources to do it by myself.

From what I can see in the crash, something is `null` where its not supposed to be. But your log seems unrelated to this crash:
```
[04Jan2022 08:00:11.501] [Server thread/ERROR] [de.geheimagentnr1.auto_restart.util.ServerRestarter/]: Auto Restart could not be done.
java.io.IOException: Cannot run program "C:\ThisPC\Desktop\MedievalMinecolonies\run.bat": CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessBuilder.start(Unknown Source) ~[?:1.8.0_311]
	at de.geheimagentnr1.auto_restart.util.ServerRestarter.restartServer(ServerRestarter.java:34) ~[auto_restart:2.0.0]
	at de.geheimagentnr1.auto_restart.handlers.ForgeEventHandler.handleServerStoppedEvent(ForgeEventHandler.java:63) ~[auto_restart:2.0.0]
	at net.minecraftforge.eventbus.ASMEventHandler_353_ForgeEventHandler_handleServerStoppedEvent_FMLServerStoppedEvent.invoke(.dynamic) ~[?:?]
	at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.fml.server.ServerLifecycleHooks.handleServerStopped(ServerLifecycleHooks.java:124) ~[forge:?]
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:683) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232) ~[?:?]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_311]
Caused by: java.io.IOException: CreateProcess error=2, The system cannot find the file specified
	at java.lang.ProcessImpl.create(Native Method) ~[?:1.8.0_311]
	at java.lang.ProcessImpl.<init>(Unknown Source) ~[?:1.8.0_311]
	at java.lang.ProcessImpl.start(Unknown Source) ~[?:1.8.0_311]
	... 11 more
```

Is there a file just named `debug.log`? Or if that one doesn't exist, check for `latest.log`.

**Comment by Zecheariah** [comment]
I'm sorry about that, new to actually making a modpack AND running the server for it.

[debug.log](https://github.com/AnonymousHacker1279/ImmersiveWeapons/files/7810803/debug.log)

**Comment by AnonymousHacker1279** [comment]
I don't see anything in there about a crash in that log, it looks like everything worked correctly.

I'll mess around a bit in a development environment to see if I can reproduce it.

**Comment by Zecheariah** [comment]
Dang, I'll set up the command block and put the command I put into and the settings as a "fix" if you don't find anything if it works fine after an hour.

Here's the modpack if you'd like to reproduce it with the additonal mods.
[[Current] ModpackAlpha 1.1-20211231T232352Z-001.zip](https://github.com/AnonymousHacker1279/ImmersiveWeapons/files/7810856/Current.ModpackAlpha.1.1-20211231T232352Z-001.zip)

**Comment by AnonymousHacker1279** [comment]
It works properly on my end.
![image](https://user-images.githubusercontent.com/48810167/148135063-29fdf1a9-7981-4c94-ba94-78d9a94bb637.png)

So that leads me to believe another mod is incompatible with Immersive Weapons. I did see a lot of hacky stuff going on in the log, so perhaps that could be related. What I will do is attempt to backport the entity spawn configurations from 1.9.3 to the 1.4.x branch, so you can at least disable the natural spawning of Hans. I don't know when I'll have this done, but I'll let you know.

**Comment by Zecheariah** [comment]
Thank you so much for all your help, right now as a temporary fix I have a command block set to "/tp @e[type=immersiveweapons:hans] ~ -64 ~"  with it running at all times unconditionally. 

As for Hans, sometimes he's fine and dandy and he spawns in just fine and then other times he's the harbinger of ticking entity.
Again thank you for your help, I really like this mod a lot!

**Comment by AnonymousHacker1279** [comment]
I've released v1.4.3 which adds configuration options to toggle the spawning of entities. Unfortunately, this really only is a band-aid fix but since it seems to be a mod conflict, I cannot do much more. But you should be able to disable Hans while keeping the other entities active, providing they aren't causing trouble as well.
